pkg/auth/spiffe_authorizer.go

```diff
@@ -0,0 +1,90 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/buildbarn/bb-storage/pkg/digest"
+	pb "github.com/buildbarn/bb-storage/pkg/proto/configuration/auth"
+	"github.com/spiffe/go-spiffe/v2/svid/x509svid"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+)
+
+// SpiffeAuthorizer  authorizes based on Spiffe-IDs
+// https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE-ID.md
+//
+// see spiffe.proto for configuration options
+type SpiffeAuthorizer struct {
```

Comment:
Be sure to make this type private, as NewSpiffeAuthorizer() can just return the interface type. This means that you don't need to document the struct and the Authorize() method. You can instead document NewSpiffeAuthorizer() (note: NewSPIFFEAuthorizer()) in lots of detail. This will then look pretty elegant if you go to pkg.go.dev/ to read the docs.